### PR TITLE
Show "delete branch" button on closed pull requests (#6570)

### LINKED ISSUE #6570: When closing a pull request, it should ask if the branch should be deleted
When closing a pull request it is easy to forget to delete the branch if it's not needed anymore.
Therefore Gitea should show the "delete branch" box which is currently only shown after a pull request has been merged.

### PR DESCRIPTION
**Pull request by saitho**
Resolves #6570 

Note that these changes introduce a new behaviour:
Closed pull requests whose pull branch was deleted can't be reopened by the user anymore.

### CODE REVIEW

**Comment by saitho** [comment]
What needs to be done in order to merge this? :)

**Comment by lunny** [comment]
@saitho need two maintainers' approvals.

File: routers/repo/pull.go
Comment (zeripath) [inline routers/repo/pull.go]:
```suggestion
	// Don't cleanup unmerged and unclosed PRs
```

**Review by zeripath — APPROVED** [review]
One small nit - I think it makes the comment clearer.

**Review by lunny — APPROVED** [review]

**Pull request merged**
